src/Config.php

```diff
@@ -3750,4 +3750,209 @@ final public static function generateUuid($type)
+            $sender = Config::getNoReplyEmailSender();
+            if ($sender['email'] !== null) {
+                return $sender;
+            } else {
+                trigger_error('No-Reply address is not defined in notifications configuration.', E_USER_WARNING);
+            }
+        }
+
+        // Try to use the configured "from" email address
+        $sender = Config::getFromEmailSender();
+        if ($sender['email'] !== null) {
+            return $sender;
+        }
+
+        // Try to use the configured "admin" email address
+        $sender = Config::getAdminEmailSender($entities_id);
+        if ($sender['email'] !== null) {
+            return $sender;
+        }
+
+        // No valid email was found, the notification will fail
+        trigger_error(
+            'No email address is not defined in notifications configuration.',
+            E_USER_WARNING
+        );
+
+        // No values found
+        return [
+            'email' => null,
+            'name'  => null,
+        ];
+    }
+
+    /**
+     * Try to find a valid "from" email from the GLPI configuration
+     * TODO: Add an entity configuration to be consistent with the behavior of
+     * getAdminEmailSender() and getReplyToEmailSender()
+     *
+     * @return array [email => sender address, name => sender name]
+     */
+    public static function getFromEmailSender(?int $entities_id = null): array
+    {
+        global $CFG_GLPI;
+
+        // Read global config data
+        $from_email      = $CFG_GLPI['from_email']      ?? "";
+        $from_email_name = $CFG_GLPI['from_email_name'] ?? "";
+
+        // Check if generic from email is valid
+        if (NotificationMailing::isUserAddressValid($from_email)) {
+            return [
+                'email' => $from_email,
+                'name'  => $from_email_name,
+            ];
+        }
+
+        // No valid values found
+        return [
+            'email' => null,
+            'name'  => null,
+        ];
+    }
+
+    /**
+     * Try to find a valid "admin_email" email to use as a sender for
+     * notifications
+     *
+     * @param int|null $entities_id Specified entity (default: current entity)
+     *
+     * @return array [email => sender address, name => sender name]
+     */
+    public static function getAdminEmailSender(?int $entities_id = null): array
+    {
+        global $CFG_GLPI;
+
+        // Check admin email in specified entity
+        $entity_admin_email = trim(
+            Entity::getUsedConfig('admin_email', $entities_id, '', '')
+        );
+        $entity_admin_email_name = trim(
+            Entity::getUsedConfig('admin_email_name', $entities_id, '', '')
+        );
+
+        // Check if admin email for the current entity is valid
+        if (NotificationMailing::isUserAddressValid($entity_admin_email)) {
+            return [
+                'email' => $entity_admin_email,
+                'name'  => $entity_admin_email_name,
+            ];
+        }
```

Comment:
We should not try to fetch entity config if `$entities_id` is null. Indeed, most of the time, sending emails is done by automated process, so it is most likely related to entity of a processed item, and not related to active entity in profile.

```suggestion
        if ($entities_id !== null) {
            $entity_admin_email = trim(
                Entity::getUsedConfig('admin_email', $entities_id, '', '')
            );
            $entity_admin_email_name = trim(
                Entity::getUsedConfig('admin_email_name', $entities_id, '', '')
            );

            // Check if admin email for the current entity is valid
            if (NotificationMailing::isUserAddressValid($entity_admin_email)) {
                return [
                    'email' => $entity_admin_email,
                    'name'  => $entity_admin_email_name,
                ];
            }
        }
```